src/compiler/output-targets/dist-lazy/lazy-output.ts

```diff
@@ -107,11 +125,45 @@ const getLazyCustomTransformer = (config: d.ValidatedConfig, compilerCtx: d.Comp
     style: 'static',
     styleImportData: 'queryparams',
   };
-  return [
-    updateStencilCoreImports(transformOpts.coreImportPath),
+  const beforeCustomTransformers = [updateStencilCoreImports(transformOpts.coreImportPath)];
+
+  if (config.transformAliasedImportPaths) {
+    beforeCustomTransformers.push(rewriteAliasedSourceFileImportPaths);
+  }
+
+  beforeCustomTransformers.push(
     lazyComponentTransform(compilerCtx, transformOpts),
-    removeCollectionImports(compilerCtx),
-  ];
+    removeCollectionImports(compilerCtx)
+  );
+  return beforeCustomTransformers;
+};
+
+/**
+ * Generate a collection of transformations that are to be applied as a part of the `afterDeclarations` step in the
+ * TypeScript compilation process.
+ *
+ * TypeScript handles the generation of JS and `.d.ts` files through different pipelines. One (possibly surprising)
+ * consequence of this is that if you modify a source file using a transformer, it will not automatically result in
+ * changes to the corresponding `.d.ts` file. Instead, if you want to, for instance, rewrite some import specifiers in
+ * both the source file _and_ its typedef you'll need to run a transformer for both of them.
+ *
+ * See here: https://github.com/itsdouges/typescript-transformer-handbook#transforms
+ * and here: https://github.com/microsoft/TypeScript/pull/23946
+ *
+ * This quirk is not terribly well documented, unfortunately.
+ *
+ * @param config the Stencil configuration associated with the current build
+ * @returns a collection of transformations that should be applied to the source code, intended for the
+ * `afterDeclarations` part of the pipeline
+ */
+const getAfterDeclarationsCustomTransformers = (
```

Comment:
question: we have this same function copy-pasted essentially. does it make sense to try to carve out a `getSharedCustomAfterDeclarationsTransformers` function that would pull out the transfomers which are common to all of our output targets? Likewise we could maybe pull out shared logic for setting up the transformers which are shared by output targets for the `before` bit. Might be more trouble than it's worth though...

Reply:
I thought about that - and I almost did it. @tanner-reits noticed that type transformations aren't working in all cases in Framework. I _think_ I'm going to end up putting the `afterDeclarations` stuff back in `run-program.ts`, but need to do some testing first before I commit to that

Reply:
Yup - removing these in https://github.com/ionic-team/stencil/pull/4501/commits/9b9eb7e3c8df296b28ea8e64b23aeecc1b6c6227